rules/aip0124/valid_reference.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0124
+
+import (
+	"fmt"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var validReference = &lint.FieldRule{
+	Name: lint.NewRuleName(124, "valid-reference"),
+	OnlyIf: func(f *desc.FieldDescriptor) bool {
+		if ref := utils.GetResourceReference(f); ref != nil {
+			urt := ref.GetType()
+			if urt == "" {
+				urt = ref.GetChildType()
+			}
+			return !stringset.New(
+				// Whitelist the common resource types in GCP.
+				// FIXME: Modularize this.
+				"cloudresourcemanager.googleapis.com/Project",
+				"cloudresourcemanager.googleapis.com/Organization",
+				"cloudresourcemanager.googleapis.com/Folder",
+				"billing.googleapis.com/BillingAccount",
```

Comment:
What about Locations?
```suggestion
				"billing.googleapis.com/BillingAccount",
				"locations.googleapis.com/Location",
```